src/transformers/models/bert_generation/tokenization_bert_generation.py

```diff
@@ -149,8 +149,17 @@ def _convert_id_to_token(self, index):
 
     def convert_tokens_to_string(self, tokens):
         """Converts a sequence of tokens (string) in a single string."""
-        out_string = self.sp_model.decode_pieces(tokens)
-        return out_string
+        current_sub_tokens = []
+        out_string = ""
+        for token in tokens:
+            # make sure that special tokens are not decoded using sentencepiece model
+            if token in self.all_special_tokens:
+                out_string += self.sp_model.decode(current_sub_tokens) + token
+                current_sub_tokens = []
+            else:
+                current_sub_tokens.append(token)
+        out_string += self.sp_model.decode(current_sub_tokens)
+        return out_string.strip()
```

Comment:
I would propose to take the same solution as the tokenizer with a fast version because if we add a fast version later we will be blocked - and the decoding method is quite important for Bert generation.

Reply:
You mean like, for example, the `convert_tokens_to_string` method in `AlbertTokenizer`?